src/main/host/descriptor/socket/inet/mod.rs

```diff
@@ -481,4 +500,25 @@ mod export {
         socket.borrow().update_packet_header(&mut packet);
         packet.into_inner();
     }
+
+    /// Decrement the ref count of the `InetSocketWeak` object. The pointer must not be used after
+    /// calling this function.
+    #[no_mangle]
+    pub extern "C" fn inetsocketweak_drop(socket: *const InetSocketWeak) {
+        assert!(!socket.is_null());
+        unsafe { Box::from_raw(socket.cast_mut()) };
+    }
+
+    /// Upgrade the weak reference. May return `NULL` if the socket has no remaining strong
+    /// references and has been dropped.
+    #[no_mangle]
+    pub extern "C" fn inetsocketweak_upgrade(socket: *const InetSocketWeak) -> *const InetSocket {
```

Comment:
I think it'd make sense for this to return a `mut` pointer, since the caller owns the returned reference

Reply:
Added.